android/src/main/java/it/innove/Peripheral.java

```diff
@@ -277,6 +286,12 @@ public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState
             readRSSICallback = null;
             registerNotifyCallback = null;
 
+            commandQueue.clear();
+            if(bleProcessing){
+              // Will set bleProcessing to false.
+              commandCompleted();
```

Comment:
It's very possible that we will get a disconnect event while a command is still processing, we need to make sure that the queue doesn't wait on that command that will never finish.